Wikipedia/Code/SavedTabBarItemProgressBadgeManager.swift

```diff
@@ -16,22 +16,33 @@
         // Observe any time a new Progress object is set. (NSProgress are not re-usable so you need to reset it if you're tracking a new progression)
         progressObjectWasSetObservation = ProgressContainer.shared.observe(\ProgressContainer.articleFetcherProgress, options: [.new, .initial]) { [weak self] (progressContainer, change) in
             self?.progressFractionCompletedObservation?.invalidate()
+            
+            var exceededMinSyncInProgressDuration = false
```

Comment:
@montehurd Is there a way to reuse this code rather than write it once for each component? 

Since both the badge and the progress bar follow the same rule, it makes sense if they are updated by the same code. It is now possible for them to get out of sync if a developer doesn't realize they have to update this code in two places.